### DISCUSSION TITLE
Programmatic access to modules/data

### DISCUSSION THREAD

**Discussion by sooheon** [bigomics/omicsplayground#144] [Q&A] [answered]
We are interested in getting the top sd or top specific gene lists for a given clustering, as well as the Louvain cluster labels out of the system. Is the only way to do this currently to load the saved data object in R?

It would be really nice to have an installable package (pgx) for programmatic use, but I’m sure it’s an issue of development hours.

**Comment by ivokwee** [comment, marked as answer]
The pgx object is actually just a regular R *rda object. You can just load the pgx file with load("mydata.pgx") and it should load an 'ngs' object. This was the old name for NGS object. The names ngs/pgx are used interchangeably. Yes, for the moment we have no API or access functions for the data slots/lists, but this would definitely be useful for the future. For the moment we did not fix yet the structure of the pgx object. In the future we may change the internal represenation toward a more standard object like SummarizedExperiment or MultiAssayExperiment. That said, the current internal respresentation has been very stable so we could think of building API (programmatic interface). Probably what is missing now, is a good documentation about the contents/structure of the PGX object.
